src/components/BannerSlider/index.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState, useCallback } from "react";
+
+interface bannerMockType {
+  img: string;
+  link: string;
+  text: string;
+}
+
+export default function BannerSlider (): JSX.Element {
+  const bannerMock: bannerMockType[] = [{
+    img: "pink",
+    link: "catalog/pink",
+    text: "Go here and buy it PINK!",
+  },
+  {
+    img: "green",
+    link: "catalog/green",
+    text: "Go here and buy it GREEN!",
+  },
+  {
+    img: "blue",
+    link: "catalog/blue",
+    text: "Go here and buy it BLUE!",
+  }];
+
+  const [bannerIndex, setBannerIndex] = useState(0);
+  const [banner, setBanner] = useState(bannerMock[0]);
+
+  const dots = Array.from({ length: bannerMock.length }, (_, i) => {
```

Comment:
Please use .map(() => ...) instead of Array.from(...).
Also put these code inside you last return JSX section.
It is redundant to create additional variable to render buttons.